src/payment/strategies/googlepay/googlepay-payment-processor.ts

```diff
@@ -54,91 +60,64 @@ export default class GooglePayPaymentProcessor {
+        return this._googlePayClient.createButton({
+            buttonColor,
+            buttonType,
             onClick,
         });
     }
 
     displayWallet(): Promise<GooglePaymentData> {
-        if (!this._googlePaymentsClient || !this._googlePaymentDataRequest || !this._googlePaymentDataRequest) {
+        if (!this._googlePayClient) {
             throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
         }
 
-        const googlePaymentsClient = this._googlePaymentsClient;
-        const googlePaymentDataRequest = this._googlePaymentDataRequest;
-
-        return this._googlePaymentsClient.isReadyToPay({
-            allowedPaymentMethods: this._googlePaymentDataRequest.allowedPaymentMethods,
+        const paymentDataRequest = this._getPaymentDataRequest();
+        const googlePayClient = this._googlePayClient;
+
+        return googlePayClient.isReadyToPay({
+            allowedPaymentMethods: [
+                {
+                    type: paymentDataRequest.allowedPaymentMethods[0].type,
+                    parameters: {
+                        allowedAuthMethods: paymentDataRequest.allowedPaymentMethods[0].parameters.allowedAuthMethods,
+                        allowedCardNetworks: paymentDataRequest.allowedPaymentMethods[0].parameters.allowedCardNetworks,
+                    },
+                },
+            ],
+            apiVersion: paymentDataRequest.apiVersion,
+            apiVersionMinor: paymentDataRequest.apiVersionMinor,
         }).then(response => {
             if (response.result) {
-                return googlePaymentsClient.loadPaymentData(googlePaymentDataRequest);
-            } else {
-                throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+                return googlePayClient.loadPaymentData(paymentDataRequest);
             }
+
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
         });
     }
 
     handleSuccess(paymentData: GooglePaymentData): Promise<InternalCheckoutSelectors> {
         return this._googlePayInitializer.parseResponse(paymentData)
             .then(tokenizedPayload => this._postForm(tokenizedPayload))
-            .then(() => this.updateBillingAddress(paymentData.cardInfo.billingAddress));
+            .then(() => this._updateBillingAddress(paymentData));
     }
 
-    private updateBillingAddress(billingAddress: GooglePayAddress): Promise<InternalCheckoutSelectors> {
-        if (!this._methodId) {
-            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
-        }
-
-        const remoteBillingAddress = this._store.getState().billingAddress.getBillingAddress();
-
-        if (!remoteBillingAddress) {
-            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+    updateShippingAddress(shippingAddress: GooglePayAddress): Promise<InternalCheckoutSelectors | void> {
+        if (!shippingAddress) {
```

Comment:
I'm not 100% sure what you're trying to do here. `shippingAddress` is a required parameter. As it cannot be `undefined`, there's no reason why you need to do a null check here. Also, if you're not passing the address in, you should throw an exception instead because, without it, you cannot update the shipping address (which is what the method is suppose to do).